browser/resources/brave_new_tab_page_refresh/components/settings/background_panel.tsx

```diff
@@ -28,38 +29,45 @@ export function BackgroundPanel() {
   const backgroundsEnabled = useAppState((s) => s.backgroundsEnabled)
   const backgroundsCustomizable = useAppState((s) => s.backgroundsCustomizable)
   const sponsoredImagesEnabled = useAppState((s) => s.sponsoredImagesEnabled)
-  const selectedBackgroundType = useAppState((s) => s.selectedBackgroundType)
   const selectedBackground = useAppState((s) => s.selectedBackground)
   const braveBackgrounds = useAppState((s) => s.braveBackgrounds)
   const customBackgrounds = useAppState((s) => s.customBackgrounds)
 
-  const [panelType, setPanelType] = React.useState<BackgroundType | null>(null)
+  const [panelType, setPanelType] =
+    React.useState(optional<SelectedBackgroundType>())
```

Comment:
This `optional` wrapper is being used (instead of `SelectedBackgroundType | null`) to help distinguish between a valid zero number value and a missing/null value in conditional expressions.

```ts
const panelType: SelectedBackgroundType | null = SelectedBackgroundType.kBrave
if (!panelType) {
  // Probably buggy
}
```

Reply:
Since there are some other usages of `Optional<T>` currently, I've created a follow-up to remove them all in one go: https://github.com/brave/brave-browser/issues/45847. I'm concerned that if I start making changes to workaround the "optional number" issue here, I'll introduce other bugs.

Reply:
Sweet, happy for this to be in a followup :smile: